deepset_cloud_sdk/workflows/pipeline_client/pipeline_service.py

```diff
@@ -0,0 +1,254 @@
+"""Pipeline publishing service for deepset Cloud SDK."""
+from typing import Protocol, runtime_checkable, AsyncGenerator, Dict, Any
+from contextlib import asynccontextmanager
+
+import structlog
+from deepset_cloud_sdk._api.deepset_cloud_api import DeepsetCloudAPI
+from deepset_cloud_sdk._api.config import CommonConfig, DEFAULT_WORKSPACE_NAME
+from deepset_cloud_sdk.workflows.pipeline_client.models import (
+    PipelineOutputs,
+    PipelineType,
+    PipelineInputs,
+    PublishConfig,
+)
+from deepset_cloud_sdk.workflows.user_facing_docs.pipeline_service_docs import PipelineServiceDocs
+
+logger = structlog.get_logger(__name__)
+
+
+@runtime_checkable
```

Comment:
This is to avoid adding Haystack to this service as a dependency.

Reply:
👍 Fair